tests/ZfrCorsTest/Factory/CorsOptionsFactoryTest.php

```diff
@@ -39,4 +39,13 @@ public function testCanCreateOptions()
 
         $this->assertInstanceOf('ZfrCors\Options\CorsOptions', $options);
     }
+
+    public function testCanCreateOptionsWithServiceManagerV2()
```

Comment:
@weierophinney updated factories like this:
```
class CorsOptionsFactory
{
    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
    {
        //...
    }
}
```
and run tests after `composer update --prefer-lowest`
```
$ php vendor/bin/phpunit 
PHPUnit v1.2.1-2-g1176b18 by Sebastian Bergmann and contributors.

Runtime:	PHP 5.6.15-1+deb.sury.org~trusty+1 with Xdebug 2.3.3
Configuration:	/vagrant/projects/zfr-cors/phpunit.xml.dist

EEE..............................

Time: 23.92 seconds, Memory: 10.50Mb

There were 3 errors:

1) ZfrCorsTest\Factory\CorsOptionsFactoryTest::testCanCreateOptions
Zend\ServiceManager\Exception\ServiceNotCreatedException: An exception was raised while creating "ZfrCors\Options\CorsOptions"; no instance returned

/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:943
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1096
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
/vagrant/projects/zfr-cors/tests/ZfrCorsTest/Factory/CorsOptionsFactoryTest.php:38

Caused by
PHPUnit_Framework_Error: Argument 3 passed to ZfrCors\Factory\CorsOptionsFactory::__invoke() must be of the type array, string given

/vagrant/projects/zfr-cors/src/ZfrCors/Factory/CorsOptionsFactory.php:39
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1096
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
/vagrant/projects/zfr-cors/tests/ZfrCorsTest/Factory/CorsOptionsFactoryTest.php:38

2) ZfrCorsTest\Factory\CorsRequestListenerFactoryTest::testCanCreateCorsRequestListener
Zend\ServiceManager\Exception\ServiceNotCreatedException: An exception was raised while creating "ZfrCors\Mvc\CorsRequestListener"; no instance returned

/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:943
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1096
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
/vagrant/projects/zfr-cors/tests/ZfrCorsTest/Factory/CorsRequestListenerFactoryTest.php:38

Caused by
PHPUnit_Framework_Error: Argument 3 passed to ZfrCors\Factory\CorsRequestListenerFactory::__invoke() must be of the type array, string given

/vagrant/projects/zfr-cors/src/ZfrCors/Factory/CorsRequestListenerFactory.php:40
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1096
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
/vagrant/projects/zfr-cors/tests/ZfrCorsTest/Factory/CorsRequestListenerFactoryTest.php:38

3) ZfrCorsTest\Factory\CorsServiceFactoryTest::testCanCreateCorsService
Zend\ServiceManager\Exception\ServiceNotCreatedException: An exception was raised while creating "ZfrCors\Service\CorsService"; no instance returned

/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:943
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1096
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
/vagrant/projects/zfr-cors/tests/ZfrCorsTest/Factory/CorsServiceFactoryTest.php:38

Caused by
PHPUnit_Framework_Error: Argument 3 passed to ZfrCors\Factory\CorsServiceFactory::__invoke() must be of the type array, string given

/vagrant/projects/zfr-cors/src/ZfrCors/Factory/CorsServiceFactory.php:39
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1096
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
/vagrant/projects/zfr-cors/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
/vagrant/projects/zfr-cors/tests/ZfrCorsTest/Factory/CorsServiceFactoryTest.php:38

FAILURES!
Tests: 33, Assertions: 59, Errors: 3.
```
so problem in V2 is here:
```
$instance = call_user_func($callable, $this, $cName, $rName);
```
third argument is string while V3 FactoryInterface expects array of options. So the simple fix would be ignoring this and removing array from __invoke signature like this:
```
public function __invoke(ContainerInterface $container, $requestedName, $options = null)
```
then tests pass. I don't know if this is ok I can change it like this.